4.gene-expression-signatures/scripts/2.run_gsea.py

```diff
@@ -0,0 +1,172 @@
+model_save_dir = pathlib.Path("saved_models")
+output_dir = pathlib.Path("gsea_results")
+output_dir.mkdir(parents=True, exist_ok=True)
+
+# Latent dimensions and model names to iterate over
+latent_dims = [2, 3, 4, 5, 6, 7, 8, 9, 10, 12, 14, 16, 18, 20, 25, 30, 35, 40, 45, 50, 60, 70, 80, 90, 100, 150, 200]
+model_names = ["pca", "ica", "nmf", "vanillavae", "betavae", "betatcvae"]
+
+final_output_file = output_dir / "combined_z_matrix_gsea_results.parquet"
+try:
+    combined_results_df = pd.read_parquet(final_output_file)
+    print(f"Loaded existing results from {final_output_file}")
+except FileNotFoundError:
+    # If the file doesn't exist, initialize an empty DataFrame
+    combined_results_df = pd.DataFrame()
+    print(f"No existing file found. Initialized empty DataFrame.")
+
+for num_components in latent_dims:
+    for model_name in model_names:
+        # Load the saved model
+        # Check if this model and latent dimension have already been processed
+        if not combined_results_df.empty:
+            if ((combined_results_df['model'] == model_name) & 
+                (combined_results_df['full_model_z'] == num_components)).any():
+                print(f"Skipping {model_name} with {num_components} dimensions as it is already processed.")
+                continue  # Skip to the next iteration if this combination is already present
+        model_filename = model_save_dir / f"{model_name}_{num_components}_components_model.joblib"
+        if model_filename.exists():
+            print(f"Loading model from {model_filename}")
+            model = joblib.load(model_filename)
+            
+            if model_name in ["pca", "ica", "nmf"]:
+                # Extract the weight matrix
+                weight_matrix_df = extract_weights(model, model_name)
+            elif model_name == "betavae":
+                weight_matrix_df = weights(model, weight_data)
+                weight_matrix_df.rename(columns={0: 'genes'}, inplace=True)
+            elif model_name == "betatcvae":
+                weight_matrix_df = tc_weights(model, weight_data)
+                weight_matrix_df.rename(columns={0: 'genes'}, inplace=True)
+            elif model_name == "vanillavae":
+                weight_matrix_df = vanilla_weights(model, weight_data)
+                weight_matrix_df.rename(columns={0: 'genes'}, inplace=True)
+            # Perform GSEA
+            gsea_results_df = perform_gsea(weight_matrix_df, model_name, num_components)
+            combined_results_df = pd.concat([combined_results_df, gsea_results_df], ignore_index=True)
+        else:
+            print(f"Model file {model_filename} not found. Skipping.")
+            
+
+
+# In[5]:
+
+
+# Save the combined dataframe to a file
+final_output_file = output_dir / "combined_z_matrix_gsea_results.parquet"
+combined_results_df.to_parquet(final_output_file, index=False)
+
+print(f"Saved final combined z_matrix and GSEA results to {final_output_file}")
+
+#Save as CSV for R 
```

Comment:
Would it make sense to port this work to Parquet (R can read Parquet files via the Arrow project).

Reply:
Yep this was just an artifact